afew/MailMover.py

```diff
@@ -50,7 +50,10 @@ def move(self, maildir, rules):
         '''
         # identify and move messages
         logging.info("checking mails in '{}'".format(maildir))
-        to_delete_fnames = []
+        # Open DB R/W if are not in dry-run
+        if not self.dry_run:
+            self.db = notmuch.Database(self.db_path,
+                                       mode=notmuch.Database.MODE.READ_WRITE)
```

Comment:
Can we set database mode inside `__init__`, as we already know there if we want run in dry-run or not?
This would avoid opening the database a second time here…

Reply:
Yes I will move this to the __init__

Reply:
So about this point, I have tested several version, but if I open the database `READ_WRITE`,
in the `__init__` the call hang after 2 requests, and get stuck, I have to kill manually.
```
$ afew --move-mails -vvvvvvvv --all
INFO:root:checking mails in 'Perso/INBOX'                                                                                                                 
DEBUG:root:query: folder:Perso/INBOX AND tag:spam AND 576540000..1522686227                                                                               
DEBUG:root:query: folder:Perso/INBOX AND tag:sent AND 576540000..1522686227
```
So maybe it comes with my own version of notmuch / Xappian but unless I find a way to make it work, I think it is better to let the DB READ_ONLY all the time and just switch to READ_WRITE when needed (like I finally did), but it would nicer to have it setup only once.

Reply:
Hanging while trying to open the database r/w a second time looks like the previous connection wasn't closed (since notmuch 0.23):

> When opening a database notmuch by default will wait for another process to release a write lock, rather than returning an error.